users.js

```diff
@@ -398,6 +398,51 @@ function exportUsergroups() {
 }
 importUsergroups();
 
+function cacheGroupData() {
+	if (Config.groups) {
+		// Support for old config groups format.
+		// Should be removed soon.
+		console.log("You are using a deprecated version of user group specification in config.\n" +
+		            "Support for this will be removed soon.\n" +
+					"Please ensure that you update your config.js to the new format (see config-example.js, line 220)\n");
+		Config.groupsranking = Config.groupsranking || Object.keys(Config.groups).reverse();
+		return false;
+	}
+	var grouplist = Config.grouplist;
+	var groups = Config.groups = Object.create(null);
+	var cachedGroups = {};
+	Config.groupsranking = [];
+
+	function cacheGroup (sym, groupData) {
+		if (cachedGroups[sym] === 'processing') return false; // cyclic inheritance.
+
+		if (cachedGroups[sym] !== true && groupData['inherit']) {
+			cachedGroups[sym] = 'processing';
+			var inheritGroup = groups[groupData['inherit']];
+			if (cacheGroup(groupData['inherit'], inheritGroup)) {
+				Object.merge(groupData, inheritGroup, false, false);
+			}
+			delete groupData['inherit'];
+		}
+		return (cachedGroups[sym] = true);
+	}
+
+	var numGroups = grouplist.length;
+	for (var i = 0; i < numGroups; i++) {
+		var groupData = grouplist[i];
+		groupData.rank = numGroups - i - 1;
+		groups[groupData.symbol] = groupData;
+		Config.groupsranking.unshift(groupData.symbol);
+	}
```

Comment:
Why don't you just make it backwards compatible? As far as I can tell, all you have to do is wrap lines 430-436 in `if (grouplist)` and move ` = Object.create(null)` from line 412 at the top of the new `if`-block.